### ISSUE TITLE
Exclude areas affecting entities, whose parent device is in excluded area

### ISSUE THREAD

**Issue by maslyankov** [nielsrowinbik/auto-sections-strategy#9]
**Describe the bug**
Exclude areas affecting entities, whose parent device is in excluded area

**To Reproduce**

```
layout: sections
views:
  - title: Main
    icon: mdi:home
    strategy:
      type: custom:auto-sections
      options:
        max_columns: 2
        group_by:
          - entity.area_id
          - area.area_id
        group_name: area.area_id|name
        filter:
          include:
            - hidden: false
          exclude:
            - area: services
        card_options:
          _:
            type: tile
```

Lets say we have a DVR in area "Services", and it has cameras as entities under that device. Some cameras are assigned another area, despite that, they are not shown, because of the exclude affecting the device and child entities.

**Expected behavior**
Hide only entities in excluded area.

**Possible solution**
Make filters more fine-grained as so:

```
layout: sections
views:
  - title: Main
    icon: mdi:home
    strategy:
      type: custom:auto-sections
      options:
        max_columns: 2
        group_by:
          - entity.area_id
          - area.area_id
        group_name: area.area_id|name
        filter:
          include:
            - entity.hidden: false
          exclude:
            - entity.area: services
        card_options:
          _:
            type: tile
```

**Comment by nielsrowinbik** [comment]
Hmm, this is strange. Maybe I've made a mistake, but [this line](https://github.com/nielsrowinbik/auto-sections-strategy/blob/8cbcf97c0de51faf692564e8d1e0f36ac1ac77b3/src/lib/utils.ts#L107), and specifically the `??` should make it so that we only fall back to the device's area if the entity's `area_id` is either `null` or `undefined`.

I just verified this within my own Home Assistant. I assigned an entity that's placed in the Bedroom through its device in the Bathroom, and it then shows up in the Bathroom section. If I then exclude the Bedroom area with an exclude filter, the entity stays (in the Bathroom section). So as far as I can tell this works precisely as intended.
